File: Luogu_Basic/b_12/p1080.py
class Person:
    def __init__(self, left, right):
        self.left = left
        self.right = right

def main():
    count = int(input())
    a = [Person(*map(int, input().split())) for _ in range(count+1)]

    a[1:] = sorted(a[1:], key=lambda p: p.left * p.right)

    ans = 0
    
    for i in range(1, count+1):
        pre = a[0].left
        for j in range(1, i):
            pre *= a[j].left
        pre = pre // a[i].right
        ans = max(ans, pre)

    print (ans)
    
main()

File: Luogu_Basic/b_12/p1080.cpp
#include <algorithm>
#include <iostream>
#include <vector>

struct person {
  long long int left;
  long long int right;
  long long int value;
};

using namespace std;

int main() {
  long long int count, max = 0;
  cin >> count;
  vector<person> a(count + 1);

  for (long long int i = 0; i < count + 1; i++) {
    cin >> a[i].left >> a[i].right;
  }

  sort(a.begin() + 1, a.end(),
       [](person c, person b) { return c.left * c.right < b.left * b.right; });

  for (long long int i = 1; i < count; i++) {
    long long int value = a[0].left;
    for (long long int j = 1; j < i; j++) {
      value *= a[j].left;
    }

    if (value > max)
      max = value;
  }
  cout << max << endl;
  return 0;
}